src/main/scala/scala/compat/java8/converterImpl/MakesSteppers.scala

```diff
@@ -5,27 +5,117 @@ import language.implicitConversions
+  def keyStepper[S <: Stepper[_]](implicit ss: StepperShape[K, S]): S with Extra
+
+  /** Generates a fresh stepper of type `S` over map values of type `V` */
+  def valueStepper[S <: Stepper[_]](implicit ss: StepperShape[V, S]): S with Extra
+}
+
+sealed trait StepperShape[T, S <: Stepper[_]] { def ref: Boolean }
+object StepperShape extends StepperShapeLowPrio {
+  private[this] def valueShape[T, S <: Stepper[_]]: StepperShape[T, S] = new StepperShape[T, S] { def ref = false }
+
+  // primitive
+  implicit val IntValue     = valueShape[Int, IntStepper]
+  implicit val LongValue    = valueShape[Long, LongStepper]
+  implicit val DoubleValue  = valueShape[Double, DoubleStepper]
+
+  // widening
+  implicit val ByteValue    = valueShape[Byte, IntStepper]
+  implicit val ShortValue   = valueShape[Short, IntStepper]
+  implicit val CharValue    = valueShape[Char, IntStepper]
+  implicit val FloatValue   = valueShape[Float, DoubleStepper]
+}
+trait StepperShapeLowPrio {
+  // reference
+  implicit def anyStepperShape[T]: StepperShape[T, AnyStepper[T]] = new StepperShape[T, AnyStepper[T]] { def ref = true }
 }
 
-trait MakesStepper[+T <: Stepper[_]] extends Any {
-  /** Generates a fresh stepper of type `T` */
-  def stepper: T
+/** Superclass for `MakesStepper` implementations which support parallelization. At least the `AnyStepper` case must be
+  * implemented, all others default to building an `AnyStepper` and putting an unboxing conversion on top. */
+trait MakesParStepper[T] extends Any with MakesStepper[T, EfficientSubstep] {
+  def stepper[S <: Stepper[_]](implicit ss: StepperShape[T, S]) = (ss match {
+    case StepperShape.IntValue    => new Stepper.UnboxingIntStepper   (stepper(StepperShape.anyStepperShape[T]).asInstanceOf[AnyStepper[Int   ]]) with EfficientSubstep
+    case StepperShape.LongValue   => new Stepper.UnboxingLongStepper  (stepper(StepperShape.anyStepperShape[T]).asInstanceOf[AnyStepper[Long  ]]) with EfficientSubstep
+    case StepperShape.DoubleValue => new Stepper.UnboxingDoubleStepper(stepper(StepperShape.anyStepperShape[T]).asInstanceOf[AnyStepper[Double]]) with EfficientSubstep
+    case StepperShape.ByteValue   => new Stepper.UnboxingByteStepper  (stepper(StepperShape.anyStepperShape[T]).asInstanceOf[AnyStepper[Byte  ]]) with EfficientSubstep
+    case StepperShape.ShortValue  => new Stepper.UnboxingShortStepper (stepper(StepperShape.anyStepperShape[T]).asInstanceOf[AnyStepper[Short ]]) with EfficientSubstep
+    case StepperShape.CharValue   => new Stepper.UnboxingCharStepper  (stepper(StepperShape.anyStepperShape[T]).asInstanceOf[AnyStepper[Char  ]]) with EfficientSubstep
+    case StepperShape.FloatValue  => new Stepper.UnboxingFloatStepper (stepper(StepperShape.anyStepperShape[T]).asInstanceOf[AnyStepper[Float ]]) with EfficientSubstep
+    case _                        => throw new NotImplementedError("AnyStepper must be handled in `stepper` implementations")
```

Comment:
I tried again and now I remember why I didn't do it this way. There are types like `BitSet` for which an `AnyStepper` doesn't make sense in the current implementation because the element type is statically known to be `Int`. Consequently there is no `AnyStepper` implementation for `BitSet`. The situation would be different if we put the `stepper` methods directly into collection classes. `BitSet`'s `stepper` methods would have to take the situation into account that a `BitSet` is for example seen as a `Set[_]`, so it would have to return an `AnyStepper[_]`, but that just means that the `stepper` implementation needs to check for the type and call `super.stepper` if it's not recognized. `BitSet` would then inherit the `AnyStepper` implementation from its parent class.